### ISSUE TITLE
"Redundant" and even wrong completion items in javascript files

### ISSUE THREAD

**Issue by bstaletic** [theia-ide/typescript-language-server#162]
Hello. This isn't a problem with typescript-language-server, but with TSserver itself and only affects javascript, not typescript.

Suppose you have this javascript file:

```javascript
const test = {
    foo: 'bar'
}
 test.
```

If you try to complete there, you'll get two completion items - `foo` and `test`. This allows users to get in a completion "loop" and do this:

    test.test.test.test.test.test.foo

This is clearly wrong and [ycmd](/ycm-core/ycmd) has been dealing with this by iterating over the candidates that tsserver returns and filtering out those that have kind equal to `warning`.

References:

- [ycmd filtering](https://github.com/ycm-core/ycmd/blob/master/ycmd/completers/typescript/typescript_completer.py#L388-L395)
- [Test project](https://github.com/ycm-core/ycmd/tree/master/ycmd/tests/javascript/testdata/identifier)
- [Test that confirms the problem](https://github.com/ycm-core/ycmd/blob/master/ycmd/tests/javascript/get_completions_test.py#L175-L199)

**Comment by DonnieWest** [comment]
Thanks for the report

This should be addressed by https://github.com/theia-ide/typescript-language-server/pull/157 though we've not released it yet. Can you confirm that it fixes the issue on your end?

**Comment by bstaletic** [comment]
I've searched the issues, so that I don't report a duplicate, but I didn't search the pull requests. I can try, but since I'm a ycmd maintainer and don't use typescript regularly, I'm having troubles figuring out how. Usually I'd just do something like [this](https://github.com/ycm-core/ycmd/blob/master/third_party/tern_runtime/package.json), but the same approach doesn't work when pulling latest master.

I figured out that `yarn` would build the project and `yarn test` runs the tests, but I can't seem to find `typescript-language-server` executable. Help would be appreciated.

**Comment by DonnieWest** [comment]
`yarn build` should do the trick

After that, you should have `server/lib/cli.js` which is the installed bin file by npm. You might need to make it executable w/ `chmod a+x` but if you point `ycmd` to that executable, you should be able to test it out

**Comment by bstaletic** [comment]
> you should have `server/lib/cli.js`

That's what I was missing! Thanks for the help. Also, there is no issue on latest master, so I'll close this thread.
